client/components/checkbox/index.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+* External dependencies
+*/
+import React from 'react';
+import classNames from 'classnames';
+import hasOwn from 'lodash/object/has';
+
+/**
+* Interal dependencies
+*/
+import Gridicon from 'components/gridicon';
+
+module.exports = React.createClass( {
+	displayName: 'Checkbox',
+
+	propTypes: {
+		onChange: React.PropTypes.func,
+		checked: React.PropTypes.any
+	},
+
+	getDefaultProps() {
+		return {
+			checked: false,
+			disabled: false
+		};
+	},
+
+	getInitialState() {
+		const props = this.props;
+		const checked = hasOwn( props, 'checked' ) ? props.checked : false;
+		return {
+			checked: checked
+		}
+	},
```

Comment:
If we make `checked` a required prop, then we should have enough protection from the developer warnings in the console to guard against having an invalid value there, which would eliminate our need for `hasOwn()` and would simplify this function

```js
getInitialState() {
	const { checked } = this.props;

	return { checked };
}
```